projects/dendrites/permutedMNIST/run_dendritic_network.py

```diff
@@ -0,0 +1,205 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2021, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+"""
+This module trains & evaluates a dendritic network in a continual learning setting on
+permutedMNIST for a specified number of tasks/permutations. A context vector is
+provided to the dendritic network, so task information need not be inferred.
+
+This setup is very similar to that of context-dependent gating model from the paper
+'Alleviating catastrophic forgetting using contextdependent gating and synaptic
+stabilization' (Masse et al., 2018).
+"""
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+
+from nupic.research.frameworks.dendrites import DendriticMLP
+from nupic.research.frameworks.pytorch.datasets import ContextDependentPermutedMNIST
+from nupic.research.frameworks.vernon import ContinualLearningExperiment, mixins
+
+
+# ------ Experiment class
+class PermutedMNISTExperiment(mixins.RezeroWeights,
+                              ContinualLearningExperiment):
+    pass
+
+
+# ------ Training & evaluation functions
```

Comment:
Are all dendrites experiment still done through this loop? As a suggestion, this could be rewritten in a few lines of code if using Vernon.

Reply:
Yes. I'm planning to merge everything with Vernon soon, but for now, there are slight modifications I needed to make to the training & evaluation functions that's easier done if I just redefine the functions themselves.